app/models/patient.rb

```diff
@@ -12,10 +13,26 @@ class Patient < ApplicationRecord
   validates :phone, presence: true
   validates :neighborhood, presence: true
 
+  after_initialize :set_main_ubs
+
+  scope :bedridden, -> { where(bedridden: true) }
+
   def active_appointments
     appointments.select(&:active?)
   end
 
+  def set_main_ubs
+    self.main_ubs ||=
+      # samples an active ubs near the patient neighborhood
+      Neighborhood.find_by(name: neighborhood)&.active_ubs&.sample ||
+      # samples any active ubs
+      Ubs.active.sample ||
+      # samples an inactive ubs near the patient neighborhood
+      Neighborhood.find_by(name: neighborhood)&.sample ||
+      # samples any inactive ubs
+      Ubs.all.sample
```

Comment:
I added these searches for inactive UBS as fallback because I had no active UBS in local environment and the application broke. I know it's an edge case that's very unlikely to happen, but anyways

Reply:
Assigning an inactive UBS like that is not the best solution IMO. We should leave the patient without an assigned UBS until some UBS is activated - then we can assign it to every patient without `main_ubs`.

Additionally, in your solution, if a neighborhood doesn't have any active UBS, the UBS assigned could potentially be inactive (even if there are active UBSs outside of that neighborhood).

Reply:
I think we should render a no UBS available message in this case. I believe it is very bad if a bedridden patient is waiting for a vaccine and there no one actually there to answer. so, I would remove these fallbacks and add an `if else` in the view. WDYT? It could say something like 'no UBS available right now', please check again later or call 'default number'.

Reply:
I think this isn't a concern right now, because we should always have at least some active `UBS`s in production. I just added these so we don't have any "loose ends" for now.